File: 프로그래머스/2/42842. 카펫/카펫.py
from math import sqrt, ceil


def solution(brown, yellow):
    br = brown + yellow

    for w in range(3, ceil(sqrt(br)) + 1):
        h = br // w

        if (w - 2) * (h - 2) == yellow:
            return sorted([w, h], reverse=True)

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int br = brown + yellow;

        for (int w = 3; w <= Math.sqrt(br); w++) {
            int h = br / w;

            if ((w - 2) * (h - 2) == yellow) {
                answer[0] = Math.max(w, h);
                answer[1] = Math.min(w, h);
                break;
            }
        }
        return answer;
    }
}